frappe/website/serve.py

```diff
@@ -0,0 +1,276 @@
+import frappe
+import os, mimetypes
+
+from werkzeug.wrappers import Response
+from werkzeug.wsgi import wrap_file
+
+from frappe.website.render import (resolve_path, build_response)
+from frappe.website.redirect import resolve_redirect
+from frappe.website.router import get_start_folders
+from frappe.website.utils import (get_home_page, can_cache, delete_page_cache,
+	get_toc, get_next_link)
+from frappe.website.doctype.website_settings.website_settings import get_website_settings
+
+def render(path=None, http_status_code=None):
+	"""render html page"""
+	if not path:
+		path = frappe.local.request.path
+
+	try:
+		path = path.strip('/ ')
+		resolve_redirect(path)
```

Comment:
not cleat what it does until resolve_redirect method is seen. why not resolve_redirect returns a redirect path and use that here?

Reply:
this is still legacy - trying to work on the core components first - then will handle the redirects